org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenSourceDownloader.java

```diff
@@ -74,6 +78,30 @@ public void discoverSource(IClassFile classFile, IProgressMonitor monitor) throw
 							artifact = new MavenCentralIdentifier().identify(path, monitor);
 						}
 						if (artifact != null) {
+							if (!ProjectUtils.isMavenProject(element.getJavaProject().getProject())) {
+								// see https://github.com/eclipse-m2e/m2e-core/commit/b547ecc358c990e182a5eaf8d36f121e43f4a8c9#diff-3967743078be6a24ba1e3ec28bfc22bdf2c88a740695411f6d20e2444fef042fR943
+								long lastModified;
+								try {
+									File artifactFile = DependencyUtil.getArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion(), artifact.getClassifier());
+									lastModified = artifactFile.lastModified();
+								} catch (FileNotFoundException | CoreException e1) {
+									lastModified = -1;
+								}
+								if (lastModified > -1) {
+									try {
+										File sources = DependencyUtil.getSources(artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion());
+										sources.setLastModified(lastModified - 1);
```

Comment:
If I understand correctly, this will force the download of sources/javadoc artifacts, but if the issue is fixed, why do we need to do this ?

Reply:
The issue is caused by https://github.com/eclipse-m2e/m2e-core/commit/b547ecc358c990e182a5eaf8d36f121e43f4a8c9#diff-3967743078be6a24ba1e3ec28bfc22bdf2c88a740695411f6d20e2444fef042fR943
It happens only for non-maven projects.

Reply:
Ok, so we force the download as before by marking the artifacts as stale. Makes sense.